src/api/llm/http.ts

```diff
@@ -72,3 +72,28 @@ export async function sendUserRating(
     throw new Error('Unexpected status code: ' + resp.status);
   }
 }
+
+export async function fetchAppNames(creds: Credentials) {
+  const response = await fetch(
+    `https://${getDomain(creds.region)}/v1/storage/files?kind=ios&kind=android`,
+    {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Basic ' + btoa(`${creds.username}:${creds.accessKey}`),
+      },
+    },
+  );
+
+  if (!response.ok) {
+    throw new Error('Network response was not ok');
```

Comment:
I don't think this will be informative enough for the user or for us if get an error report. You see it in the error message at https://github.com/saucelabs/vscode-scriptiq/blob/f70bbdbdf2371309a1f7e17aae1d4b3f9f37dabe/src/editors/TestGenerationPanel.ts#L179-L179 as well, where you have to make a guess about what could be wrong. We'd definitely know if we get an unauthorized response back. I wouldn't hide the error, just show it to the user.